### PR TITLE
update rest api to reflect case-server changes

### PR DESCRIPTION
**Pull request by AbdelHedhili**
Signed-off-by: AbdelHedhili <abdelsalem.hedhili@rte-france.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -49,23 +49,24 @@
...
+                                                                  @PathVariable("caseUuid") UUID caseUuid,
                                                                   @RequestParam("description") String description) {
 
         if (studyService.studyExists(studyName)) {
             throw new ResponseStatusException(HttpStatus.CONFLICT, STUDY_ALREADY_EXISTS);
         }
 
-        if (!studyService.caseExists(caseName)) {
-            throw new ResponseStatusException(HttpStatus.NOT_FOUND, CASE_DOESNT_EXISTS);
+        if (!studyService.caseExists(caseUuid)) {
```
Comment (jonenst) [inline src/main/java/org/gridsuite/study/server/StudyController.java:65]: why 'not' caseExists ?

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 2]
```diff
@@ -49,23 +49,24 @@
...
+            throw new ResponseStatusException(HttpStatus.CONFLICT, CASE_DOESNT_EXISTS);
```
Comment (jonenst) [inline src/main/java/org/gridsuite/study/server/StudyController.java:66]: keep NOT_FOUND ?

Comment (AbdelHedhili) [reply src/main/java/org/gridsuite/study/server/StudyController.java:66]: of course don't know what went through my mind...

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 3]
```diff
@@ -236,14 +247,14 @@ public void test() throws Exception {
...
+        mvc.perform(post("/v1/studies/{studyName}/cases/{caseUuid}", STUDY_NAME, caseUuid)
                 .param(DESCRIPTION, DESCRIPTION))
                 .andExpect(status().isOk());
 
         //insert a study with a non existing case and except exception
-        result = mvc.perform(post("/v1/studies/{studyName}/cases/{caseName}", "randomStudy", "notExistingCase")
+        result = mvc.perform(post("/v1/studies/{studyName}/cases/{caseUuid}", "randomStudy", "00000000-0000-0000-0000-000000000000")
                 .param(DESCRIPTION, DESCRIPTION))
-                .andExpect(status().isNotFound())
+                .andExpect(status().isConflict())
```
Comment (jonenst) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:257]: isNotFound () ?

File: src/main/java/org/gridsuite/study/server/StudyConstants.java [hunk 4]
```diff
@@ -23,6 +23,7 @@ private StudyConstants() {
 
     static final String CASE_NAME = "caseName";
```
Comment (jonenst) [inline src/main/java/org/gridsuite/study/server/StudyConstants.java:24]: remove this one ?

**Pull request merged**